protocol/x/subaccounts/types/position_size.go

```diff
@@ -1,19 +1,26 @@
 package types
 
 import (
-	errorsmod "cosmossdk.io/errors"
 	"math/big"
 
+	errorsmod "cosmossdk.io/errors"
+
 	"github.com/dydxprotocol/v4-chain/protocol/dtypes"
 )
 
+const (
+	AssetProductType     = "asset"
+	PerpetualProductType = "perpetual"
+)
+
 // PositionSize is an interface for expressing the size of a position
 type PositionSize interface {
 	// Returns true if and only if the position size is positive.
 	GetIsLong() bool
 	// Returns the signed position size in big.Int.
 	GetBigQuantums() *big.Int
 	GetId() uint32
+	GetProductType() string
```

Comment:
Used in `checkPositionUpdatable` which is generic for `AssetUpdate` and `PerpetualUpdate`